js/plugins/vertexai/src/imagen.ts

```diff
@@ -106,7 +153,15 @@ interface ImagenInstance {
   image?: { bytesBase64Encoded: string };
 }
 
-export function imagen2Model(client: GoogleAuth, options: PluginOptions) {
+export function imagenModel(
+  name: string,
+  client: GoogleAuth,
+  options: PluginOptions
+) {
+  const modelName = `vertexai/${name}`;
+  const model: ModelReference<z.ZodTypeAny> = SUPPORTED_IMAGEN_MODELS[name];
+  if (!model) throw new Error(`Unsupported model: ${name}`);
+
   const predictClients: Record<
```

Comment:
I'm not changing this, but this code was curious to me. Does this actually provide any value? It appears to be intended as some type of cache/memoization, but is it actually providing that function?

Reply:
It's caching clients per location (location can be passes in the request)... why do it you think it may not be doing that?

Reply:
I think it just looks funny, coming from more of a pure OO background. Thanks.